low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -0,0 +1,467 @@
+            elif state == "active":
+                # inactive/failed -> active
+                if service in self.failed_services:
+                    self.failed_services.remove(service)
+                    alert_info_index = 2
+                if service in self.not_active_services:
+                    self.not_active_services.pop(service)
+            elif state == "failed":
+                # inactive -> failed
+                if service not in self.failed_services:
+                    self.failed_services.append(service)
+                    alert_info_index = 0
+            else:
+                alert_info_index = 3
+        elif prev_state == "activating":
+            if service in self.not_active_services:
+                self.not_active_services.pop(service)
+            if state in ["inactive", "deactivating"]:
+                # activating -> inactive/deactivating
+                self.failed_services.append(service)
+                alert_info_index = 0
+            elif state == "active":
+                # activating -> active
+                if service in self.failed_services:
+                    self.failed_services.remove(service)
+                    alert_info_index = 2
+                else :
+                    # its a restart.
+                    pass
+            elif state == "failed":
+                # activating -> failed
+                if service not in self.failed_services:
+                    self.failed_services.append(service)
+                    alert_info_index = 0
+            else:
+                alert_info_index = 3
+
+        if alert_info_index != -1:
+            self.raise_alert(service, prev_state, state,
+                prev_substate, substate, prev_pid, pid,
+                alert_info_index)
+
+    def raise_alert(self, service, prev_state, state, prev_substate,
+                    substate, prev_pid, pid, alert_info_index):
+        """ Send the alert to ServiceMsgHandler."""
+        # Each alert info contains 4 fields
+        # 1.Description | 2.Alert Type | 3.Impact | 4.Recommendation
+        alert_info = [
+            [f"{service} in {state} state.",                #index 0
+                "fault",
+                f"{service} is not available for use.",
+                "Try to restart the service"],
+            [f"{service} in a non_active state for more than {self.max_wait_time} seconds.",
+                "fault",                                     #index 1
+                f"{service} is not available for use.",
+                "Try to restart the service"],
+            [f"{service} in {state} state.",
+                "fault_resolved",                            #index 2
+                f"{service} is available for use now.",
+                ""],
+            [f"service state transition from {prev_state} to {state} is not handled.",
+                "missing",                                   #index 3
```

Comment:
missing? Should be 'informational' as default

Reply:
Then we will have to add something to this list. Since any other alert_type does not match..
ALERT_TO_SEVERITY_MAPPING = {
         "fault": "critical",
         "fault_resolved": "informational",
         "missing": "critical",
         "insertion": "informational",
         "threshold_breached:low": "warning",
         "threshold_breached:high": "warning"
    }

Reply:
removed the alert.